### PR TITLE
Updating pavlova example to Dart 2.

### PR DESCRIPTION
**Pull request by Sfshaza**
Please review, @Hixie. 

Still need to convert the rest of the examples/tutorial.

https://sz-flutter-2.firebaseapp.com/tutorials/layout/#nesting

### CODE REVIEW

**Comment by Sfshaza** [comment]
@cbracken @Hixie @mit-mit, do any of you have the bandwidth to review this?

thx

File: _includes/code/layout/pavlova/main.dart [hunk 1]
```diff
@@ -38,47 +38,47 @@ class MyHomePage extends StatefulWidget {
...
+        children: <Widget>[
           new Row(
             mainAxisSize: MainAxisSize.min,
-            children: [
-              new Icon(Icons.star, color: Colors.black),
-              new Icon(Icons.star, color: Colors.black),
-              new Icon(Icons.star, color: Colors.black),
-              new Icon(Icons.star, color: Colors.black),
-              new Icon(Icons.star, color: Colors.black),
+            children: <Widget>[
```
Comment (Hixie) [inline _includes/code/layout/pavlova/main.dart]: i'm surprised the analyzer didn't suggest the list itself could be const.

**Comment by Hixie** [comment]
LGTM, but I would assume that many of those type annotations aren't actually necessary if your goal is to be as brief as possible.

**Comment by Hixie** [comment]
What is the status of this PR?

File: _includes/code/layout/pavlova/main.dart [hunk 2]
```diff
@@ -38,47 +38,47 @@ class MyHomePage extends StatefulWidget {
 class _MyHomePageState extends State<MyHomePage> {
   @override
   Widget build(BuildContext context) {
-    var titleText = new Container(
-      padding: new EdgeInsets.all(20.0),
-      child: new Text(
+    final Container titleText = new Container(
```
Comment (leafpetersen) [inline _includes/code/layout/pavlova/main.dart]: I don't know about flutter specific lints, but this type is redundant with inference unless Container is a generic class (in which case I would still get rid of the RHS type and make it new Container<dynamic>(.

File: _includes/code/layout/pavlova/main.dart [hunk 3]
```diff
@@ -38,47 +38,47 @@ class MyHomePage extends StatefulWidget {
...
+        style: const TextStyle(
           fontWeight: FontWeight.w800,
           letterSpacing: 0.5,
           fontSize: 30.0,
         ),
       ),
     );
 
-    var subTitle = new Text(
+    final Text subTitle = const Text(
```
Comment (leafpetersen) [inline _includes/code/layout/pavlova/main.dart]: Redundant as above.

File: _includes/code/layout/pavlova/main.dart [hunk 4]
```diff
@@ -38,47 +38,47 @@ class MyHomePage extends StatefulWidget {
...
-      style: new TextStyle(
+      style: const TextStyle(
         fontFamily: 'Georgia',
         fontSize: 25.0,
       ),
     );
 
-    var ratings = new Container(
-      padding: new EdgeInsets.all(20.0),
+    final Container ratings = new Container(
```
Comment (leafpetersen) [inline _includes/code/layout/pavlova/main.dart]: redundant as above.

File: _includes/code/layout/pavlova/main.dart [hunk 5]
```diff
@@ -38,47 +38,47 @@ class MyHomePage extends StatefulWidget {
...
+      padding: const EdgeInsets.all(20.0),
       child: new Row(
         mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-        children: [
+        children: <Widget>[
```
Comment (leafpetersen) [inline _includes/code/layout/pavlova/main.dart]: redundant with inference.

File: _includes/code/layout/pavlova/main.dart (hunk 1 above)
Comment (leafpetersen) [inline _includes/code/layout/pavlova/main.dart]: redundant <Widget> with inference

File: _includes/code/layout/pavlova/main.dart [hunk 6]
```diff
@@ -90,7 +90,7 @@ Pavlova is a meringue-based dessert named after the Russian ballerina Anna Pavlo
       ),
     );
 
-    var descTextStyle = new TextStyle(
+    final TextStyle descTextStyle = const TextStyle(
```
Comment (leafpetersen) [inline _includes/code/layout/pavlova/main.dart]: redundant.

File: _includes/code/layout/pavlova/main.dart [hunk 7]
```diff
@@ -101,43 +101,43 @@ Pavlova is a meringue-based dessert named after the Russian ballerina Anna Pavlo
 
     // DefaultTextStyle.merge allows you to create a default text
     // style that is inherited by its child and all subsequent children.
-    var iconList = DefaultTextStyle.merge(
+    final Widget iconList = DefaultTextStyle.merge(
```
Comment (leafpetersen) [inline _includes/code/layout/pavlova/main.dart]: redundant

File: _includes/code/layout/pavlova/main.dart [hunk 8]
```diff
@@ -101,43 +101,43 @@ Pavlova is a meringue-based dessert named after the Russian ballerina Anna Pavlo
...
       style: descTextStyle,
       child: new Container(
-        padding: new EdgeInsets.all(20.0),
+        padding: const EdgeInsets.all(20.0),
         child: new Row(
           mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-          children: [
+          children: <Widget>[
```
Comment (leafpetersen) [inline _includes/code/layout/pavlova/main.dart]: all of the <Widget> in this are, I believe, redundant.

File: _includes/code/layout/pavlova/main.dart [hunk 9]
```diff
@@ -101,43 +101,43 @@ Pavlova is a meringue-based dessert named after the Russian ballerina Anna Pavlo
...
               ],
             ),
           ],
         ),
       ),
     );
 
-    var leftColumn = new Container(
-      padding: new EdgeInsets.fromLTRB(20.0, 30.0, 20.0, 20.0),
+    final Container leftColumn = new Container(
```
Comment (leafpetersen) [inline _includes/code/layout/pavlova/main.dart]: redundant

File: _includes/code/layout/pavlova/main.dart [hunk 10]
```diff
@@ -146,7 +146,7 @@ Pavlova is a meringue-based dessert named after the Russian ballerina Anna Pavlo
       ),
     );
 
-    var mainImage = new Image.asset(
+    final Image mainImage = new Image.asset(
```
Comment (leafpetersen) [inline _includes/code/layout/pavlova/main.dart]: redundant

File: tutorials/layout/index.md [hunk 11]
```diff
@@ -896,24 +896,24 @@ class _MyHomePageState extends State<MyHomePage> {
   Widget build(BuildContext context) {
     //...
 
-    var ratings = new Container(
-      padding: new EdgeInsets.all(20.0),
+    final Container ratings = new Container(
```
Comment (leafpetersen) [inline tutorials/layout/index.md]: redundant

File: tutorials/layout/index.md [hunk 12]
```diff
@@ -950,7 +950,7 @@ class _MyHomePageState extends State<MyHomePage> {
   Widget build(BuildContext context) {
     // ...
 
-    var descTextStyle = new TextStyle(
+    final TextStyle descTextStyle = const TextStyle(
```
Comment (leafpetersen) [inline tutorials/layout/index.md]: redundant

File: tutorials/layout/index.md [hunk 13]
```diff
@@ -896,24 +896,24 @@ class _MyHomePageState extends State<MyHomePage> {
...
     //...
 
-    var ratings = new Container(
-      padding: new EdgeInsets.all(20.0),
+    final Container ratings = new Container(
+      padding: const  EdgeInsets.all(20.0),
       child: new Row(
         mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-        children: [
+        children: <Widget>[
```
Comment (leafpetersen) [inline tutorials/layout/index.md]: redundant

File: tutorials/layout/index.md [hunk 14]
```diff
@@ -961,32 +961,32 @@ class _MyHomePageState extends State<MyHomePage> {
 
     // DefaultTextStyle.merge allows you to create a default text
     // style that is inherited by its child and all subsequent children.
-    var iconList = DefaultTextStyle.merge(
+    final Widget iconList = DefaultTextStyle.merge(
```
Comment (leafpetersen) [inline tutorials/layout/index.md]: redundant

File: tutorials/layout/index.md [hunk 15]
```diff
@@ -1009,10 +1009,10 @@ class _MyHomePageState extends State<MyHomePage> {
   Widget build(BuildContext context) {
     //...
 
-    var leftColumn = new Container(
-      padding: new EdgeInsets.fromLTRB(20.0, 30.0, 20.0, 20.0),
+    final Container leftColumn = new Container(
```
Comment (leafpetersen) [inline tutorials/layout/index.md]: redundant

**Comment by Sfshaza** [comment]
OK, I've had discussions with the team about which analyzer options and which lints we should be using. I'm closing this PR.

**Pull request closed without merging**